### ISSUE TITLE
[repository] Provide a namespace for categorizing resources

### ISSUE THREAD

**Issue by laeubi** [osgi/osgi#602]
Currently the [Repository Service Specification](https://docs.osgi.org/specification/osgi.cmpn/8.0.0/service.repository.html) defines an [osgi.content Namespace](https://docs.osgi.org/specification/osgi.cmpn/8.0.0/service.repository.html#i3224340).

But a repository sometimes do not only contains a loose collection of resources but has some mean of categorizing/grouping items. This is especially common when installing from an UI where a user wants to navigate through possible candidate items, a popular example is when you install new software from a P2 software site:

![grafik](https://github.com/osgi/osgi/assets/1331477/20415b20-d6a0-4f82-b315-958ab9a9471c)

I therefore like to propose a new namespace provided by the repository specification that allows to categorize items.

```
public class RepositoryCategoryNamespace {

    /**
     * Namespace name for repository category capabilities and requirements.
     */
    public static final String REPOSITORY_CATEGORY_NAMESPACE = "osgi.repository.category";

    /**
     * The capability attribute identifying a name for this category.
     */
    public static final String CAPABILITY_NAME_ATTRIBUTE = "name";

    /**
     * The capability attribute identifying the description of the category.
     */
    public static final String CAPABILITY_DESCRIPTION_ATTRIBUTE = "description";
}
```

A category can then be modeled by a resources that requires all its child items.

**Comment by timothyjward** [comment]
As discussed on the call, I can see why this is a compelling thing to put in a repository.

A couple of comments:

- Is a category really a namespace? Namespaces define requirements and capabilities, but a category doesn't seem like either of those things. Should it be a resource type?
- Who will the users of this standard be? If this is intended to provide category information in the Eclipse installer would it be sufficient to use an Eclipse specific type?

I'm pretty convinced that the answer to the first question is no, but I don't really know about the second. If the set of users is wider than Eclipse then standardising may make sense, if not then the existing extensibility in the repository definition would allow this to work without specification changes

**Comment by timothyjward** [comment]
> Is a category really a namespace? Namespaces define requirements and capabilities, but a category doesn't seem like either of those things. Should it be a resource type?

Yes - it is a resource with an identity capability and a `type`. Attributes `name` and `description` (already in use) provide the human readable text.

**Comment by maho7791** [comment]
I could see a this as kind of tagging a resource to a certain category

**Comment by laeubi** [comment]
Reusing the identity seems good, it might still be good to describe the type in the Repository spec even though it is more a "User Display Thing", e.g. I can think about a webrenderer of a repository that shows you the content of a `repository.xml` in a human friendly way that also displays categories.

Also [Knopflerfish](https://www.knopflerfish.org/) has an UI for the user and also BNDTools show things to the user and might benefit.

**Comment by laeubi** [comment]
Still an issue...
